lib/ext/fhir/library.rb

```diff
@@ -2,14 +2,24 @@
 module FHIR
   class Library < DomainResource
 
-    def create_data_elements(fhir_value_sets)
+    def create_data_elements(value_sets)
       dataRequirement.map do |data_req|
         type = data_req.type.value
-        oid = data_req.codeFilter.first&.valueSet&.value&.match(/([0-2])((\.0)|(\.[1-9][0-9]*))*$/).to_s
-        fhir_value_set = fhir_value_sets.find{ |fvs| fvs.fhirId == oid }
-        title = fhir_value_set&.name&.value
+        oid, title =
+          # for ValueSets
+          if data_req.codeFilter&.first&.valueSet?
+            oid = data_req.codeFilter.first&.valueSet&.value&.match(/([0-2])((\.0)|(\.[1-9][0-9]*))*$/).to_s
+            value_set = value_sets.find{ |vs| vs.fhirId == oid }
+            title = value_set&.name&.value
+            [oid, title]
+          # for DRCs
+          elsif data_req.codeFilter&.first&.code?
+            title = data_req.codeFilter.first&.code&.first&.display&.value
```

Comment:
Shell we use code value If the display value is missing? Just in case..

Reply:
👍

Reply:
would there be a case where code is without display?

Reply:
After a few attempts in the MAT, I was unable to generate a DRC without the `display` value in the bundle.

The `display` value in the FHIR Bundle comes from CQL code identifier clause, which is required by the MAT, and not the CQL code display clause, which is optional.

Reply:
that was my understanding as well. never saw code without label in MAT. Maybe I can remove the safe navigation &